### PR TITLE
feat(language-selection-screen): UX & settingsSlice 

### PR DESCRIPTION
**Pull request by VladislavRiabko**
- add LanguageSelection screen 
- add settingsSlice 
- add setLanguage action 
- refactor MainHeader component 
- refactor RadioButtonList component

### CODE REVIEW

File: app/assets/images/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export const IMAGES = {
   LOGO: require('./beyondCodelineLogo.png'),
   WHITE_PENCIL: require('./whitePencil.png'),
+  WHITE_BACK_ARROM: require('./whiteBackArrow.png'),
```
Comment (emberglaz) [inline app/assets/images/index.ts]: ``WHITE_BACK_ARROW: require('./whiteBackArrow.png'),

File: app/redux/slices/settingsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+interface ISettingsState {
+  language: string;
+}
+
+const INITIAL_STATE: ISettingsState = {
+  language: 'en',
+};
+
+export const settingsSlice = createSlice({
+  name: 'user',
```
Comment (emberglaz) [inline app/redux/slices/settingsSlice.ts]: `name: 'settings',`

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { IMAGES } from '../../../../assets';
+
+import {
+  BottomButtonView,
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: refactor spacelines

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  MainHeader,
+  RadioButtonList,
+  RadioButtonListType,
+} from '../../../../components';
+import { useAppDispatch } from '../../../../hooks';
+import { settingsSlice } from '../../../../redux/slices/settingsSlice';
+import i18n from '../../../../services/localization';
+import { generalStyles } from '../../../../utils/styles';
+import { LANGUAGES } from './LanguageSelection.data';
+import { ILanguageSelectionScreenProps } from './LanguageSelection.props';
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: refactor

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+export const LanguageSelectionScreen: React.FC<ILanguageSelectionScreenProps> =
+  ({ navigation }) => {
+    const disptach = useAppDispatch();
+
+    const [correctLanguage, setCorrectLanguage] = useState<string[]>(['en']);
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: rename ->`selectedLanguage`

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    const { t } = useTranslation();
+
+    useEffect(() => {
+      i18n.changeLanguage(correctLanguage[0]);
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: the same

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      disptach(
+        settingsSlice.actions.setLanguage({ language: correctLanguage[0] }),
+      );
+    }, [correctLanguage, disptach]);
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: remove disptach

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+    return (
+      <SafeAreaView style={generalStyles.whFlex}>
+        <BottomButtonView buttonTitle={t('buttons.confirm')} onSubmit={_.noop}>
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx:39]: add `// TODO: `

**Review by emberglaz — CHANGES_REQUESTED** [review]

File: app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { SafeAreaView } from 'react-native-safe-area-context';
```
Comment (emberglaz) [inline app/screens/Onboarding/Parents/LanguageSelection/LanguageSelection.tsx]: import from rn

**Review by emberglaz — CHANGES_REQUESTED** [review]

File: app/components/RadioButtonList/RadioButtonList.props.ts [hunk 10]
```diff
@@ -6,4 +6,5 @@ import {
 export interface IRadioButtonListProps {
   data: IRadioButtonListItem[];
   type: RadioButtonListType;
+  setSelected: (item: string[]) => void;
```
Comment (Reykjavik151) [inline app/components/RadioButtonList/RadioButtonList.props.ts]: item but array type. so rename it to `items`

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**